src/main/java/io/github/bbortt/event/planner/service/ResponsibilityService.java

```diff
@@ -81,4 +90,37 @@ public void delete(Long id) {
         log.debug("Request to get all Responsibilities for Project {}", projectId);
         return responsibilityRepository.findAllByProjectId(projectId, pageable);
     }
+
+    /**
+     * Checks if the current user has access to the `Project` linked to the given `Responsibility`, identified by id. The project access must be given by any of the `roles`. Example usage: `@PreAuthorize("@responsibilityService.hasAccessToResponsibility(#responsibility, 'ADMIN', 'SECRETARY')")`
+     *
+     * @param responsibilityId the id of the responsibility with a linked project to check.
+     * @param roles to look out for.
+     * @return true if the project access has any of the roles.
+     */
+    @PreAuthorize("isAuthenticated()")
+    public boolean hasAccessToResponsibility(Long responsibilityId, String... roles) {
+        Optional<Responsibility> responsibility = findOne(responsibilityId);
+        return responsibility.map(value -> hasAccessToResponsibility(value, roles)).orElse(true);
```

Comment:
I return `true` as default, so the corresponding resource method will map to `HTTP 404` instead of `HTTP 403`. I have no possibility to determine the `Project` if the `Responsibility` doesn't exist. do you think `HTTP 403` would be more appropriate?

Reply:
Würde eher HTTP 403 zurückgeben.